src/noisepy/seis/utils.py

```diff
@@ -0,0 +1,19 @@
+import os
+import posixpath
+from urllib.parse import urlparse
+
+import fsspec
+
+S3_SCHEME = "s3"
+
+
+def get_filesystem(path: str) -> fsspec.AbstractFileSystem:
+    url = urlparse(path)
+    return fsspec.filesystem(S3_SCHEME, anon=True) if url.scheme == S3_SCHEME else fsspec.filesystem("file")
```

Comment:
```suggestion
def get_filesystem(path: str, storage_options: dict = {}) -> fsspec.AbstractFileSystem:
    url = urlparse(path)
    if url.scheme == S3_SCHEME:
        storage_options = {'anon': True}
    return fsspec.filesystem(url.scheme, **storage_options)
```

Just a little change to this code. Since this is a "utility" module, I think it should be agnostic in terms of file system, but defaulting to the storage option of `anon=True` for s3. However, maybe you'd like to use this utility again for other file system such as Google Cloud or Azure Blob Storage / Data Lake.

Reply:
Thanks, I'll make a small update, add the `anon=True`  only if the the `anon` key is not already there.